.github/workflows/R-CMD-check.yaml

```diff
@@ -0,0 +1,104 @@
+# NOTE: This workflow is overkill for most R packages
+# check-standard.yaml is likely a better choice
+# usethis::use_github_action("check-standard") will install it.
+#
+# For help debugging build failures open an issue on the RStudio community with the 'github-actions' tag.
+# https://community.rstudio.com/new-topic?category=Package%20development&tags=github-actions
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+name: R-CMD-check
+
+jobs:
+  R-CMD-check:
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macOS-latest,   r: 'release'}
+          - {os: windows-latest, r: 'release'}
+          - {os: windows-latest, r: '3.6'}
+          - {os: ubuntu-16.04,   r: 'devel', rspm: "https://packagemanager.rstudio.com/cran/__linux__/xenial/latest", http-user-agent: "R/4.0.0 (ubuntu-16.04) R (4.0.0 x86_64-pc-linux-gnu x86_64 linux-gnu) on GitHub Actions" }
```

Comment:
I know the default is ubuntu 16... why not 18 / bionic to help avoid updating when 16 is removed in the far future?

Reply:
Yea, Jeroen and Jim had an interesting back-and-forth on #r-lib about this the other day (Jim's in the 16.04 all the way camp). In thematic, I recently changed devel to 20.04, and release to 18.04...I'd be game for doing the same here

Reply:
Up to you which you choose.  Just make sure to use the `matrix.config.os` value when installing dependencies.  Also be sure to use the `matrix.config.os` value in your cache path